index.rst

```diff
@@ -207,8 +207,8 @@ When a filename is given to EditorConfig a search is performed in the
 directory of the given file and all parent directories for an EditorConfig
 file. An EditorConfig file is named ".editorconfig", all lowercased.
 Non-existing directories are treated as if they exist and are empty. All found
-EditorConfig files are searched for sections with section names matching the
-given filename. The search shall stop if an EditorConfig file is found with
+EditorConfig files are searched for sections with section glob expressions matching 
```

Comment:
This is a different topic than "empty editorconfig files."  Would you please move it to a separate PR?  That would make the git history cleaner, and this PR more focused.  Thanks!

Reply:
Fully agree. I can move it into a separate "polishing" commit. What are you thoughts on this, @cxw42 ?

CC: @xuhdev

Reply:
Actually, now that I think about it more, I believe this should stay as it was.  "Section name" is defined above ([here](https://spec.editorconfig.org/#id8)).  Using the term in this paragraph is therefore unambiguous.

Consider the section name `[{*.txt,*.md}]`.  At the command prompt, I would think of that as two "glob expression"s (`*.txt` and `*.md`).  However, in this paragraph the whole name `{...}` is taken as a unit.  I worry that if we made this change it might confuse readers.  Sorry for the late change!